package.json

```diff
@@ -31,6 +31,7 @@
     "@angular/core": "^4.0.0",
     "@angular/forms": "^4.0.0",
     "@angular/http": "^4.0.0",
+    "@angular/router": "^4.0.0",
```

Comment:
I don't think that's the case. When I was making `@angular/http` optional for the icons, I had the same approach and removing the entire modules from the `node_modules` still worked up until the point where we actually hit the  HTTP module functionality. Since we're only referencing the type, not the class directly, TypeScript doesn't need to load the module. [Here's an example that mirrors what we have in this PR.](https://www.typescriptlang.org/play/index.html#src=import%20%7B%20Router%2C%20OtherRouterClass%20%7D%20from%20'%40angular%2Frouter'%3B%0D%0A%0D%0Aexport%20class%20Something%20%7B%0D%0A%20%20%20%20constructor(private%20router%3A%20Router)%20%7B%0D%0A%20%20%20%20%20%20%20%20router.doStuff()%3B%0D%0A%20%20%20%20%7D%0D%0A%7D) Note how the `define` doesn't reference `@angular/router`, that will only happen if you try to do something like `new Router()`.